rust/connlib/tunnel/src/lib.rs

```diff
@@ -119,6 +119,8 @@ where
             )? {
                 Poll::Ready(io::Input::Timeout(timeout)) => {
                     self.role_state.handle_timeout(timeout);
+                    self.callbacks
+                        .on_update_resources(self.role_state.resources());
```

Comment:
So this is when some heartbeat times out and something may have gone offline, so we update the resources list?

Reply:
due to a timeout things can go unknown not offline but yeah, we do this because after the timeouts something may have changed state and we should report.